src/main/groovy/net/vivin/gradle/versioning/VersionUtils.groovy

```diff
@@ -117,6 +113,14 @@ class VersionUtils {
                 return headTag - TAG_PREFIX_PATTERN
             }
         }
+
+        if(version.snapshot) {
+            result = "$result-$version.config.snapshotSuffix"
+        } else if (versionByTag.containsValue(result)) {
+            throw new BuildException("Determined version '$result' already exists in the repository at '$repository.directory'.\nFix your bumping or manually create a tag with the intended version on the commit to be released.", null)
```

Comment:
Well, actually it could also be that the settings for filtering are wrong or whatever. 
But maybe the message should be rewritten, as it shouldn't say what you thought it said.
The intention was, bumping settings produced 1.2.3 as bumpPatch is set, which is already present.
Either set bumpMinor to get 1.3.0 or manually tag the commit with 1.3.0 before building.
It was not meant to advice the user to produce an ambiguous tag, hence the "intended version" in the message.